checker/check_result.go

```diff
@@ -92,3 +92,27 @@ func MakeProportionalResult(name string, numerator int, denominator int,
 		Confidence: MaxResultConfidence - int(actual*MaxResultConfidence),
 	}
 }
+
+// MultiCheckAnd means all checks must succeed. This returns a conservative result
+// where the worst result is returned.
+func MakeAndResult(checks ...CheckResult) CheckResult {
+	minResult := CheckResult{
+		Pass:       true,
+		Confidence: MaxResultConfidence,
+	}
+
+	for _, result := range checks {
```

Comment:
can you also patch function [MultiCheckAnd()](https://github.com/ossf/scorecard/blob/main/checker/check_runner.go#L109) to return `MakeAndResult`?